conformance/lang/expressions/xml-attribute-access-expr/xml-required-attribute-access-expr.balt

```diff
@@ -0,0 +1,456 @@
+
+function errorFunction() {
+    xml x = xml `<message id="10">
+                        <note id="11">
+                            <to>Tove</to>
+                        </note>
+                        <note id="12">
+                            <to>Jani</to>
+                        </note>
+                    </message>`;
+    _ = x.<messages>.id; // @error incompatible types: expected 'any', found '(string|error)'
+    _ = x.<message>/<notes>[0].id; // @error incompatible types: expected 'any', found '(string|error)'
+    _ = x.<message>/<notes>[1].id; // @error incompatible types: expected 'any', found '(string|error)'
+}
+
+Test-Case: error
+Description: Test xml-required-attribute-access-expr when attribute name does not exist.
+Labels: string, xml, xml-filter-expr, xml-required-attribute-access-expr, xml-step-expr
+
+function errorFunction() {
+    xml x = xml `<message id="10">
+                        <note id="11">
+                            <to>Tove</to>
+                        </note>
+                        <note id="12">
+                            <to>Jani</to>
+                        </note>
+                    </message>`;
+    _ = x.<message>.ids; // @error error("{ballerina/lang.xml}XMLOperationError",message="attribute 'ids' not found")
+    _ = x.<message>/<note>[0].ids; // @error error("{ballerina/lang.xml}XMLOperationError",message="attribute 'ids' not found")
+    _ = x.<message>/<note>[1].ids; // @error error("{ballerina/lang.xml}XMLOperationError",message="attribute 'ids' not found")
+}
+
+Test-Case: output
+Description: The static type of an xml-required-attribute-access-expr is string|error The static type of an xml-optional-attribute-access-expr is string|error|().
+Labels: error-type, nil-type, is-expr, string, xml, xml-required-attribute-access-expr, xml-optional-attribute-access-expr
+
+function init() {
+    xml x = xml `<root attr='attr-val'></root>`;
+    io:println(x.attr is string); // @output true
+    io:println(x.attrs is error); // @output true
+    io:println(x?.attr is string); // @output true
+    io:println(x?.attrs is ()); // @output true
+    xml y = xml `xml text=text`;
+    io:println(y?.text is error); // @output true
+}
+
+Test-Case: error
+Description: Test xml-attribute-access-expr when static type of expression is not a subtype of xml.
+Labels: xml, xml-required-attribute-access-expr, xml-optional-attribute-access-expr, int, string, xml:Text, union-type
+
+function errorFunction() {
+    xml|int x1 = xml `<root attr="attr-val"></root>`;
+    _ = x1.attr; // @error invalid operation: type '(xml|int)' does not support field access
+    xml|xml:Text x2 = xml `<root attr="attr-val"></root>`;
+    _ = x2.attr; // @error invalid operation: type '(xml|xml:Text)' does not support field access
+    _ = x2?.attr1; // @error invalid operation: type '(xml|xml:Text)' does not support optional field access
+}
```

Comment:
Shall we add tests to cover invalid attribute access on other subtypes of XML?
Need to consider https://github.com/ballerina-platform/ballerina-lang/issues/35191 too.